SeQuant/core/rational.hpp

```diff
@@ -33,12 +34,31 @@ std::wstring to_wstring(const boost::rational<T>& i) {
 }  // namespace boost
 
 namespace sequant {
-using rational = boost::rational<std::int64_t>;
+
+// introducing multiprecision
+using namespace boost::multiprecision;
+
+// typedef boost::multiprecision::number<
+//     cpp_int_backend<0, 0, signed_magnitude, checked, void>>
+//     mp_int_type;
+
+// try high fixed precision
+typedef int512_t mp_int_type;
```

Comment:
```suggestion
using int512_t = mp_int_type;
```
(that's the more c++/modern way of doing it)